### DISCUSSION TITLE
osm-tags-filter without specific tag

### DISCUSSION THREAD

**Discussion by StephanGeorg** [kraina-ai/quackosm#48] [Q&A] [answered]
Hey @RaczeQ.

Filtering by available tag works quite well 

```bash
quackosm DE.pbf --osm-tags-filter='{"addr:place":true}' 
```
but the combination of

```bash
quackosm DE.pbf --osm-tags-filter='{"addr:place":true,"addr:street":false}'

```
does not seem to work. Expected output would be entities with tag `addr:place` and without `addr:street`. 
Thanks for a hint.

**Comment by RaczeQ** [comment]
Hi @StephanGeorg, currently it doesn't work like that at all, but I can see where this could be a useful feature (and might be confusing).

The filtering logic in QuackOSM is based on OSMnx, where only a `True` value is considered in a filtering process.

You can see it here:
https://github.com/gboeing/osmnx/blob/cabdbb246dc14095f5d8e8b8ead38f9649dae2c7/osmnx/features.py#L1029C13-L1035C54
https://github.com/gboeing/osmnx/blob/cabdbb246dc14095f5d8e8b8ead38f9649dae2c7/osmnx/_overpass.py#L226

This way was probably done to simplify Overpass queries generated by the OSMnx when querying a server. Here we can utilize negative values in the filter since we have the raw data available, but that would require some overhaul for generating SQL queries executed in DuckDB. I'll move it to a new issue for the future.

For now, I'd probably use `DuckDB`'s query to quickly scan the entire dataset and load all feature IDs beforehand and use them as a filter in QuackOSM. 
Another option would be to load all the data and filter it afterwards with `DuckDB` or `pyarrow.compute` module.
```python
import duckdb
import quackosm as qosm

duckdb.load_extension('spatial')

# Load all the OSM elements from the PBF file with 'addr:place' and without 'addr:street'
features_df = duckdb.sql("""
SELECT
    kind || '/' || id as feature_id
FROM
    ST_READOSM('DE.pbf')
WHERE
    tags IS NOT NULL
    AND list_contains(map_keys(tags), 'addr:place')
    AND NOT list_contains(map_keys(tags), 'addr:street')
""").to_df()

features_ids = list(features_df.feature_id)

# gpq_path = qosm.convert_pbf_to_gpq(file_paths="DE.pbf", filter_osm_ids=features_ids)
gdf = qosm.get_features_gdf(file_paths="DE.pbf", filter_osm_ids=features_ids)
```

**Comment by RaczeQ** [comment, marked as answer]
This feature is now available since version `0.6.0` 😉 
#49 #53
